powerapi-core/src/main/scala/org/powerapi/core/power/Power.scala

```diff
@@ -23,132 +23,165 @@
+    case KILOWATTS => bounded(max_kw)
+    case MEGAWATTS => bounded(max_Mw)
+    case _ =>
+      val v = MEGAWATTS.convert(value, unit)
+      0.0 <= v && v <= max_Mw
+  }, "Power value is limited to 1.79e308 mW and cannot be negative")
+
+  def toKiloWatts: Double = unit.toKiloWatts(value)
+
+  def toMegaWatts: Double = unit.toMegaWatts(value)
+
+  def toUnit(u: PowerUnit): Double = toMilliWatts / MILLIWATTS.convert(1, u)
+
+  override def toString(): String = s"$value $unit"
+
+  def compare(other: Power): Int = toMilliWatts compare other.toMilliWatts
+
+  def toMilliWatts: Double = unit.toMilliWatts(value)
+
+  def +(other: Power): Power = add(other.value, other.unit)
+
+  private[this] def add(otherValue: Double, otherUnit: PowerUnit): Power = {
+    val commonUnit = if (otherUnit.convert(1, unit) < 1.0) unit else otherUnit
+    val resultValue = safeAdd(commonUnit.convert(value, unit), commonUnit.convert(otherValue, otherUnit))
+    new RawPower(resultValue, commonUnit)
+  }
+
   private[this] def safeAdd(a: Double, b: Double): Double = {
     if ((b > 0.0) && (a > Double.MaxValue - b)) throw new IllegalArgumentException("double overflow")
     if ((b < 0.0) && (a < -b)) throw new IllegalArgumentException("negative power cannot exists")
     a + b
   }
-  private[this] def add(otherValue: Double, otherUnit: PowerUnit): Power = {
-    val commonUnit = if (otherUnit.convert(1, unit) < 1.0) unit else otherUnit
-    val resultValue = safeAdd(commonUnit.convert(value, unit), commonUnit.convert(otherValue, otherUnit))
-    new RawPower(resultValue, commonUnit)
+
+  def -(other: Power): Power = add(-other.value, other.unit)
+
+  def *(factor: Double): Power = {
+    if (factor.isInfinite || factor.isNaN) {
+      throw new IllegalArgumentException("factor must be a finite and defined value")
+    }
+    else RawPower(safeMul(value * factor), unit)
   }
 
-  def +(other: Power) = add(other.value, other.unit)
-  def -(other: Power) = add(-other.value, other.unit)
-  
   private[this] def safeMul(a: Double): Double = {
     if (a.isInfinite) throw new IllegalArgumentException("multiplication's result is an infinite value")
     if (a.isNaN) throw new IllegalArgumentException("multiplication's result is an undefined value")
     if (a > Double.MaxValue) throw new IllegalArgumentException("double overflow")
     if (a < 0.0) throw new IllegalArgumentException("negative power cannot exists")
     a
   }
-  
-  def *(factor: Double) = new RawPower({
-      if (factor.isInfinite || factor.isNaN) throw new IllegalArgumentException("factor must be a finite and defined value")
-      else safeMul(value * factor)
-    }, unit
-  )
-  def /(divisor: Double) = new RawPower({
-      if (divisor.isInfinite || divisor.isNaN) throw new IllegalArgumentException("divisor must be a finite and defined value")
-      else safeMul(value / divisor)
-    }, unit
-  )
-  
-  override def equals(other: Any) = other match {
+
+  def /(divisor: Double): Power = {
```

Comment:
[Scalastyle] WARNING: Method name does not match the regular expression '^[a-z][A-Za-z0-9]*(_=)?$'